packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,33 +1,292 @@
+            break
+          } else {
+            eHideRoot = hideRoot
+          }
+        }
+      }
+      return {
+        visibleItems: [...currentVisibleItems],
+        menuItems: [...currentMenuItems],
+        effectiveHideRoot: eHideRoot,
+      }
+    },
+    [
+      MIN_VISIBLE_ITEMS,
+      childArray,
+      childArrayWidths,
+      effectiveHideRoot,
+      hideRoot,
+      overflow,
+      rootItemWidth,
+      menuButtonWidth,
+    ],
+  )
+
+  const handleResize = useCallback(
+    (entries: ResizeObserverEntry[]) => {
+      if (entries[0]) {
+        const containerWidth = entries[0].contentRect.width
+        const result = calculateOverflow(containerWidth)
+        setVisibleItems(result.visibleItems)
+        setMenuItems(result.menuItems)
+        setEffectiveHideRoot(result.effectiveHideRoot)
+      }
+    },
+    [calculateOverflow],
+  )
+
+  useResizeObserver(handleResize, containerRef)
+
+  // Initial overflow calculation for testing and >5 items
+  useEffect(() => {
+    if (overflow === 'menu' && childArray.length > 5) {
+      // Get actual container width from DOM or use default
+      const containerWidth = containerRef.current?.offsetWidth || 800
+      const result = calculateOverflow(containerWidth)
+      setVisibleItems(result.visibleItems)
+      setMenuItems(result.menuItems)
+      setEffectiveHideRoot(result.effectiveHideRoot)
+    }
+  }, [overflow, childArray.length, calculateOverflow])
+
+  // Determine final children to render
+  const finalChildren = React.useMemo(() => {
+    if (overflow === 'wrap' || menuItems.length === 0) {
+      return React.Children.map(children, child => <li className={classes.ItemWrapper}>{child}</li>)
+    }
+
+    let effectiveMenuItems = [...menuItems]
+    if (!effectiveHideRoot) {
+      effectiveMenuItems = [...menuItems.slice(1)]
+    }
+    const menuElement = (
+      <li className={classes.BreadcrumbsItem} key="breadcrumbs-menu">
+        <BreadcrumbsMenuItem
+          ref={menuButtonRef}
+          items={effectiveMenuItems}
+          aria-label={`${effectiveMenuItems.length} more breadcrumb items`}
+        />
+        <ItemSeparator />
+      </li>
+    )
+
+    const visibleElements = visibleItems.map((child, index) => (
+      <li className={classes.BreadcrumbsItem} key={`visible + ${index}`}>
+        {child}
+        <ItemSeparator />
+      </li>
+    ))
+
+    const rootElement = (
+      <li className={classes.BreadcrumbsItem} key={`rootElement`}>
+        {rootItem}
+        <ItemSeparator />
+      </li>
+    )
+
+    // Position menu based on effective hideRoot setting and visible items
+    if (effectiveHideRoot) {
+      // Show: [overflow menu, leaf breadcrumb]
+      return [menuElement, ...visibleElements]
+    } else {
+      // Show: [root breadcrumb, overflow menu, leaf breadcrumb]
+      return [rootElement, menuElement, ...visibleElements]
+    }
+  }, [overflow, menuItems, effectiveHideRoot, visibleItems, rootItem, children])
```

Comment:
If I'm understanding right, I think on every resize this value will be memoized is that right? Even if the items aren't changing. Would it make sense in `handleResize` to compare the two lists (visible items, menu items) to see if they have changed before updating?

I'm also curious if, with React compiler, we need to use memo for this 🤔 Curious what you've run into for that 👀